### PR TITLE
NSURLSession: Delay the temporary file deletion until delegate completion handler is invoked.

### PR DESCRIPTION
**Pull request by msft-Jeyaram**
Tested with AFNetworking sample and we are able to download and display the images. #1047

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1670)

### CODE REVIEW

File: Frameworks/Foundation/NSURLSession.mm [hunk 1]
```diff
@@ -761,15 +757,21 @@ - (void)downloadTask:(NSURLSessionDownloadTask*)task
...
         [inflightTaskData dispatchDownloadCompletionHandlerForURL:url];
+        [self _destroyTemporaryFile:url];
     } else {
         // NOT OPTIONAL (protocol specifies @required)
         auto properDelegate(reinterpret_cast<id<NSURLSessionDownloadDelegate>>(_delegate));
         [_delegateQueue addOperationWithBlock:^{
             [properDelegate URLSession:self downloadTask:task didFinishDownloadingToURL:url];
+            [self _destroyTemporaryFile:url];
         }];
     }
```
Comment (msft-Jeyaram) [inline Frameworks/Foundation/NSURLSession.mm]: the changes.

looks like precommit clang script  decided to format it it's way

File: Frameworks/Foundation/NSURLSession.mm [hunk 2]
```diff
@@ -761,15 +757,21 @@ - (void)downloadTask:(NSURLSessionDownloadTask*)task
                              totalBytesExpectedToWrite);
 }
 
+- (void)_destroyTemporaryFile:(NSURL*)url {
```
Comment (msft-Jeyaram) [inline Frameworks/Foundation/NSURLSession.mm]: verified via AFNetworking sample.

thank you @memontic-ms for the sample

File: Frameworks/Foundation/NSURLSession.mm [hunk 3]
```diff
@@ -761,15 +757,21 @@ - (void)downloadTask:(NSURLSessionDownloadTask*)task
...
+    [[NSFileManager defaultManager] removeItemAtURL:url error:nullptr];
```
Comment (aballway) [inline Frameworks/Foundation/NSURLSession.mm]: nit: nil

Comment (DHowett-MSFT) [inline Frameworks/Foundation/NSURLSession.mm]: yes, nil

---
In reply to: [95426628](https://github.com/Microsoft/WinObjC/pull/1670#discussion_r95426628) [](ancestors = 95426628)

**Comment by DHowett-MSFT** [comment]
:shipit:

File: Frameworks/Foundation/NSURLSession.mm (hunk 3 above)
Comment (DHowett-MSFT) [inline Frameworks/Foundation/NSURLSession.mm]: wait a second, no. not nil! aj you tricked me!

it's an NSError**, so, nullptr.

---
In reply to: [95448529](https://github.com/Microsoft/WinObjC/pull/1670#discussion_r95448529) [](ancestors = 95448529,95426628)

**Pull request merged**